langchain4j/src/main/java/dev/langchain4j/store/embedding/WeaviateEmbeddingStore.java

```diff
@@ -0,0 +1,98 @@
+package dev.langchain4j.store.embedding;
+
+import dev.langchain4j.data.embedding.Embedding;
+import dev.langchain4j.data.segment.TextSegment;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.List;
+import lombok.Builder;
+
+public class WeaviateEmbeddingStore implements EmbeddingStore<TextSegment> {
+
+  private final EmbeddingStore<TextSegment> implementation;
+
+  @Builder
+  public WeaviateEmbeddingStore(String apiKey, String scheme, String host, String objectClass) {
+    try {
+      implementation =
+        loadDynamically(
```

Comment:
BTW what do you think in general about this dynamic loading approach? The idea was to have a `WeaviateEmbeddingStore` class in the main library so that it is visible to the users of the library that there is such an integration. Not everyone reads documentation (which we dont even have yet 😂)... But the dynamic loading part is super hacky. Do you have an idea to how make it less hacky?

Reply:
I'm new to this dynamic loading part and yes, I find it also very hacky 😄 since I'm new to it, I have no optimization ideas yet  😄